src/GovUk.Education.ExploreEducationStatistics.Admin/Services/UserManagementService.cs

```diff
@@ -448,10 +448,13 @@ public async Task<Either<ActionResult, Unit>> DeleteUser(string email)
                         .Where(invite => invite.Email.ToLower() == email.ToLower())
                         .ToListAsync();
 
-                    // Delete the internal EES user, if found.
+                    // We don't remove the ContentDb's Users entry. All other stuff related to the
+                    // user is removed. This is done so if that user's Id exists in other table's
```

Comment:
I think you have missed deleting `UserPublicationRole` and `UserReleaseRole` entries.

They extend from `ResourceRole` which has a `User` attribute.

Methods like this https://github.com/dfe-analytical-services/explore-education-statistics/blob/1b7bb64f7649b33fb895639065592a54e9af8b7e/src/GovUk.Education.ExploreEducationStatistics.Admin/Services/UserReleaseRoleRepository.cs#L91 are returning data from those tables joined with the ContentDb's user.

I can see a bug where views like the Publication team access tab are listing users which have been deleted.